Assets/MixedRealityToolkit.SDK/Experimental/HandCoach/Scripts/MoveToTarget.cs

```diff
@@ -0,0 +1,127 @@
+
+        [Tooltip("Lerp curve")]
+        [SerializeField]
+        private AnimationCurve m_animationCurve = AnimationCurve.Linear(0f, 0f, 1f, 1f);
+
+        // The local position of this gameObject
+        private Vector3 m_position;
+
+        // The local position of m_targetObject
+        private Vector3 m_targetPosition;
+
+        // The offset from this gameObject's position to the m_trackingObject's position
+        private Vector3 m_trackingOffset;
+
+        // bool to determine when to stop the follow sequence
+        private bool m_followingTargetObject;
+
+        private void Awake()
+        {
+            if (m_trackingObject != null)
+            {
+                m_trackingOffset = new Vector3(0,0,0); // GetRelativeLocalPosition(m_trackingObject, m_rootObject) - gameObject.transform.parent.localPosition;
+            }
+        }
+
+        // Since this script can attach to an object with an animator, we need to update position in LateUpdate
+        private void LateUpdate()
+        {
+            if (m_targetObject != null && m_rootObject != null)
+            {
+                m_targetPosition = GetRelativeLocalPosition(m_targetObject, m_rootObject) + m_targetOffset;
+                transform.parent.localPosition = m_position;
+            }
+        }
+
+        // Starts coroutine to lerp from current position to target position
+        public void MoveToTargetPosition()
+        {
+            if (m_targetPosition != null)
+            {
+                m_followingTargetObject = false;
+                StartCoroutine(MoveHintSequence());
+            }
+        }
+
+        private IEnumerator MoveHintSequence()
+        {
+            Vector3 origin = m_position;
+
+            float t = 0;
+            while (t <= m_duration)
+            {
+                m_position = Vector3.Lerp(origin, m_targetPosition, m_animationCurve.Evaluate(t / m_duration));
+                t += Time.deltaTime;
+                yield return null;
+            }
+        }
+
+        public void SetHintTarget(GameObject target)
+        {
+            m_targetObject = target;
+        }
+
+        // Starts coroutine to follow the target object
+        public void Follow()
+        {
+            if (m_trackingObject != null && m_rootObject != null)
+            {
+                m_followingTargetObject = true;
+                StartCoroutine(FollowSequence());
+            }
+        }
+
+        private IEnumerator FollowSequence()
+        {
+            while (m_followingTargetObject)
+            {
+                m_position = GetRelativeLocalPosition(m_trackingObject, m_rootObject) - m_trackingOffset;
+                yield return null;
+            }
+        }
+
+        private Vector3 GetRelativeLocalPosition(GameObject input, GameObject root)
+        {
+            Vector3 sum = new Vector3();
+            Transform currentLevel = input.transform;
+            while (currentLevel.gameObject != root)
```

Comment:
couldn't you just do input.transform.position - root.transform.position
if you substract world pos of root from the input world pos you should end up with the relative position of the input gameobject to the root gameobject

Reply:
If the returned vector needs to be in local space you can use `root.transform.InverseTransformPoint(input.transform.position)`

Reply:
That's much simpler, changed.